pex/resolver.py

```diff
@@ -925,7 +935,7 @@ def iter_targets():
           yield DistributionTarget.for_platform(platform)
 
   download_request = DownloadRequest(
-    targets=list(iter_targets()),
+    targets=OrderedSet(iter_targets()),
```

Comment:
Using an `OrderedSet` isn't only an optimization, but necessary to avoid the duplicate requests, right? If so, consider adding a brief N.B. comment explaining why it's important that we de-duplicate.

Reply:
Correct and done.